exorcist/models.py

```diff
@@ -1,23 +1,31 @@
 from enum import Enum
-from dataclasses import dataclass
-from datetime import datetime
+from typing import TypeVar, Generic
+import dataclasses
+
+# generics: the actual types here depend on the client library
+Result = TypeVar("Result")
+TaskDetails = TypeVar("TaskDetails")
 
 class TaskStatus(Enum):
+    """
+    Status of a given task.
+    """
     BLOCKED = 0
     AVAILABLE = 1
     IN_PROGRESS = 2
     RESULTS_READY = 3
     COMPLETED = 99
+    TOO_MANY_RETRIES = -2
     ERROR = -1
 
 
-@dataclass
-class Task:
-    taskid: str
-    taskfile: str
-    last_modified: datetime
-    n_retries: int
+# TODO: it isn't entirely clear to me that this is needed, or that this is
+# the right way to do it. but I wanted to capture the way to handle typing
+# of something like this
+@dataclasses.dataclass
+class Task(Generic[TaskDetails]):
```

Comment:
this class is going right?

Reply:
removed in [240b325](https://github.com/OpenFreeEnergy/exorcist/pull/9/commits/240b3255297a57ba79900a462789673a14b58595)